deepchem/trans/transformers.py

```diff
@@ -654,14 +655,46 @@ def realize(self, similarity, y, w):
     """
     features = []
     similarity_xs = similarity * np.sign(w)
-    for similarity_x in similarity_xs:
-      pair = list(zip(similarity_x, range(len(similarity_x))))
-      pair.sort(key=lambda x: x[0], reverse=True)
-      if pair[0][0] >= 1:
-        pair = pair[1:self.K + 1]
+    [target_len, reference_len] = similarity_xs.shape
+    g_temp = tf.Graph()
+    values = []
+    top_labels = []
+    with g_temp.as_default():
+      labels_tf = tf.constant(y)
+      similarity_placeholder = tf.placeholder(
+          dtype=tf.float64, shape=(None, reference_len))
+      value, indice = tf.nn.top_k(
+          similarity_placeholder, k=self.K + 1, sorted=True)
+      # the tf graph here pick up the (K+1) highest similarity values 
+      # and their indices
+      top_label = tf.gather(labels_tf, indice)
+      # map the indices to labels
+      feed_dict = {}
+      for count in range(target_len // 100 + 1):
+        feed_dict[similarity_placeholder] = similarity_xs[count * 100:min((
+            count + 1) * 100, target_len), :]
+        # generating batch of data by slicing similarity matrix 
+        # into 100*reference_dataset_length
+        with tf.Session() as sess:
```

Comment:
A new session is being made every step in the for-loop. This will be quite slow. Could you move the session creation outside it?